### PR TITLE
Fix `lr_find` to generate same results on multiple calls

### LINKED ISSUE #9672: lr_find not resetting Trainer model
## 🐛 Bug

If `tuner.lr_find` is called multiple times on the same trainer the observed behavior is as if the tuner didn't restore the parameters upon finishing, suggesting that the trainer is maintaining a reference to the modified parameters instead of to the restored ones. 

This may have been the cause of issue #6675. 

This is pure speculation but this may as well be an issue with `tuner.scale_batch_size` leading to an apparent memory leak (see #9625)

### To Reproduce

Simple example:
```python
#init model
#init optimizer
#init trainer
for n in [100, 200, 300]:
    #trainer should be instantiated here for correct behaviour
    lr_finder = trainer.tuner.lr_find(model, ... , num_training=n)
    # do something with lr_finder
```
[BoringModel](https://gist.github.com/grudloff/a302e5778d246926d84a1a5b333d84c9)

Here 2 runs of lr_find are run, once with no reintantiation of the trainer in test_bad() and once with reinstantiation of the trainer in test_good(). Here are the curves for both cases:
image
image
As can be seen, when there is no reinstantiation it seems the second run starts from the previous run's end, while on the second case both curves are the same.

@tchaton

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #9672

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: tests/tuner/test_lr_finder.py [hunk 1]
```diff
@@ -283,3 +284,15 @@ def training_step_end(self, outputs):
...
+def test_multiple_lr_find_calls_gives_same_results(tmpdir):
+    """Tests that lr_finder gives same results if called multiple times."""
+    dm = ClassifDataModule()
+    model = ClassificationModel()
+
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=2)
+    all_res = [trainer.tuner.lr_find(model, datamodule=dm).results for _ in range(3)]
+
+    for k in all_res[0].keys():
+        assert all(all_res[0][k] == curr_lr_finder[k] for curr_lr_finder in all_res[1:])
```
Comment (awaelchli) [inline tests/tuner/test_lr_finder.py:298]:
```suggestion
    model = BoringModel()

    trainer = Trainer(default_root_dir=tmpdir, max_epochs=2)
    all_res = [trainer.tuner.lr_find(model).results for _ in range(3)]

    for k in all_res[0].keys():
        assert all(all_res[0][k] == curr_lr_finder[k] for curr_lr_finder in all_res[1:])
```

there is no need to use a complex model, we can use the boring model. 
test fails on master.

**Review by awaelchli — APPROVED** [review]

File: tests/tuner/test_lr_finder.py (hunk 1 above)
Comment (rohitgr7) [reply tests/tuner/test_lr_finder.py:298]: sometimes `lr_find` fails when using `BoringModel` since it doesn't converge and it just logs the exception. So I prefer to use a model with lr_finder tests that at least converge to confirm that it works as expected.

Comment (awaelchli) [reply tests/tuner/test_lr_finder.py]: don't we want to set a seed then? I  mean just initially for the test
